core/domain/question_services_test.py

```diff
@@ -312,9 +340,11 @@ def test_get_question_summaries_by_creator_id(self):
             question_services.get_question_summaries_by_creator_id(
                 self.editor_id))
 
-        self.assertEqual(len(question_summaries), 1)
+        questions = [self.question_id, self.question_id_1, self.question_id_2]
+
+        self.assertEqual(len(question_summaries), 3)
         for question_summary in question_summaries:
-            self.assertEqual(question_summary.id, self.question_id)
+            self.assertTrue(question_summary.id in questions)
```

Comment:
This is weaker, you need to add an assertion for no duplicates if you do this. Or maybe find a way to compare them in sorted order.

Reply:
Done. I added an assertion for no duplicates because the order is random.

Reply:
Here and above, it is better if you can sort them and compare equality. Is that possible?

(Looking at how you're checking no-duplicates, it's a bit clumsy because you need to micromanage all the indices. The code should ideally be more general.)

Reply:
Done.